crates/cubecl-wgpu/src/compiler/wgsl/compiler.rs

```diff
@@ -10,25 +10,28 @@ use cubecl_core::{
     ir::{self as cube, HybridAllocator, UIntKind},
     prelude::CompiledKernel,
     server::ComputeServer,
-    Feature,
+    Feature, Metadata,
 };
 use cubecl_runtime::{DeviceProperties, ExecutionMode};
-use wgpu::{ComputePipeline, DeviceDescriptor, ShaderModuleDescriptor};
+use wgpu::{
+    BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingType, BufferBindingType,
+    ComputePipeline, DeviceDescriptor, PipelineLayoutDescriptor, ShaderModuleDescriptor,
+    ShaderStages,
+};
 
 /// Wgsl Compiler.
 #[derive(Clone, Default)]
 pub struct WgslCompiler {
     num_inputs: usize,
     num_outputs: usize,
+    metadata: Metadata,
```

Comment:
So the `Metadata` type is acting like the metadata of the metadata 😅. Maybe we could name the type `MetadataHeader` ?

Reply:
It represents the metadata conceptually on the compiler side, so compilers don't need to do their own offset calculations. So I think the name is appropriate. It's the dynamic equivalent of `offset_of!`.